packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -0,0 +1,191 @@
+                    fieldKind: fieldChange.fieldKind,
+                    change: brand(rebasedField),
+                };
+            }
+        }
+
+        return rebasedFields;
+    }
+
+    rebaseAnchors(anchors: AnchorSet, over: FieldChangeMap): void {
+        anchors.applyDelta(this.intoDelta(over));
+    }
+
+    intoDelta(change: FieldChangeMap): Delta.Root {
+        const delta: Delta.Root = new Map();
+        for (const field of Object.keys(change)) {
+            const fieldChange = change[field];
+            const deltaField = this.getChangeHandler(fieldChange.fieldKind).intoDelta(
+                fieldChange.change,
+                (childChange) => this.intoDelta(childChange),
+            );
+            delta.set(brand(field), deltaField);
+        }
+        return delta;
+    }
+
+    buildEditor(deltaReceiver: (delta: Delta.Root) => void, anchors: AnchorSet): ModularEditBuilder {
+        return new ModularEditBuilder(this, deltaReceiver, anchors);
+    }
+
+    encodeForJson(formatVersion: number, change: FieldChangeMap): JsonCompatibleReadOnly {
+        const encodedFields: { [key: string]: JsonCompatibleReadOnly; } = {};
+        for (const field of Object.keys(change)) {
+            const fieldChange = change[field];
+            encodedFields[field] = this.getChangeHandler(fieldChange.fieldKind).encoder.encodeForJson(
+                formatVersion,
+                fieldChange.change,
+                (childChange) => this.encodeForJson(formatVersion, childChange),
+            );
+        }
+
+        return encodedFields;
+    }
+
+    encodeBinary(formatVersion: number, change: FieldChangeMap): Buffer {
+        throw new Error("Method not implemented.");
+    }
+
+    decodeJson(formatVersion: number, change: JsonCompatibleReadOnly): FieldChangeMap {
+        const encodedChange = change as Record<string, JsonCompatibleReadOnly>;
+        const decodedFields: FieldChangeMap = {};
+        for (const field of Object.keys(encodedChange)) {
+            const fieldChange = encodedChange[field] as unknown as EncodedFieldChange;
+            const fieldChangeset = this.getChangeHandler(fieldChange.fieldKind).encoder.decodeJson(
+                formatVersion,
+                fieldChange.change,
+                (encodedChild) => this.decodeJson(formatVersion, encodedChild),
+            );
+
+            decodedFields[field] = { fieldKind: fieldChange.fieldKind, change: brand(fieldChangeset) };
+        }
+
+        return decodedFields;
+    }
+
+    decodeBinary(formatVersion: number, change: Buffer): FieldChangeMap {
+        throw new Error("Method not implemented.");
+    }
+
+    private getChangeHandler(kind: FieldKindIdentifier): FieldChangeHandler<unknown> {
+        const fieldKind = this.schemaPolicy.fieldKinds.get(kind);
+        assert(fieldKind !== undefined, "Unknown field kind");
+        return fieldKind.changeHandler;
+    }
+}
+
+interface EncodedFieldChange {
```

Comment:
it seems odd that this is used in decode, and not used in the encode codepath. Maybe an explicit type annotation using it could be added to help ensure encoded data matches this format?

Reply:
Good suggestion. This actually caught a bug with the encoder.